[LWG motion 16] P1901R2 Enabling the Use of weak_ptr as Keys in Unordered Associative… (#6361)
cplusplus/draft: source/memory.tex

\begin{codeblock}
namespace std {
  struct owner_equal {
    template<class T, class U>
      bool operator()(const shared_ptr<T>&, const shared_ptr<U>&) const noexcept;
    template<class T, class U>
      bool operator()(const shared_ptr<T>&, const weak_ptr<U>&) const noexcept;
    template<class T, class U>
      bool operator()(const weak_ptr<T>&, const shared_ptr<U>&) const noexcept;
    template<class T, class U>
      bool operator()(const weak_ptr<T>&, const weak_ptr<U>&) const noexcept;

    using is_transparent = @\unspec@;
  };
}
\end{codeblock}

\indexlibrarymember{operator()}{owner_equal}%
\begin{itemdecl}
template<class T, class U>
  bool operator()(const shared_ptr<T>& x, const shared_ptr<U>& y) const noexcept;
template<class T, class U>
  bool operator()(const shared_ptr<T>& x, const weak_ptr<U>& y) const noexcept;
template<class T, class U>
  bool operator()(const weak_ptr<T>& x, const shared_ptr<U>& y) const noexcept;
template<class T, class U>
  bool operator()(const weak_ptr<T>& x, const weak_ptr<U>& y) const noexcept;
\end{itemdecl}

\begin{itemdescr}
\pnum
\returns
\tcode{x.owner_equal(y)}.

\pnum
\begin{note}
\tcode{x.owner_equal(y)} is \tcode{true}

> %FIXME: Who are "they"?
> if and only if they share ownership or are both empty.

Reviewer: Looking at the added `shared_ptr::owner_equal` and `weak_ptr::owner_equal`,
```suggestion
if and only if \tcode{x} and \tcode{y} share ownership or are both empty.
```
